### ISSUE TITLE
systemd: serial-getty@.service should support 57600 baud

### ISSUE THREAD

**Issue by mbiebl** [systemd/systemd#17306]
Version: 246.6
Forwarding this verbatim from the downstream bug report at https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=969144

Dear Maintainer,

My BIOS provides pre-boot access to the system over serial, and sets
the port up for 57600.  So I have configured the client (conserver)
to use that speed.  This has happened to work by accident with the
systemd serial-getty@.service because of the  --keep-baud
parameter to agetty.

But today I ended up changing the speed of the port while temporarily
testing another client.  And was surprised to learn that it is
impossible to switch back to 57600 with the default serial-getty@.service 

I see no reason why this standard baud rate should be excluded from
the list of rates.  Please fix

```
- --- a/lib/systemd/system/serial-getty@.service   2020-04-27 19:02:57.000000000 +0200
+++ b/etc/systemd/system/serial-getty@.service   2020-08-28 09:45:03.000000000 +0200
@@ -31,7 +31,7 @@
 # The '-o' option value tells agetty to replace 'login' arguments with an
 # option to preserve environment (-p), followed by '--' for safety, and then
 # the entered username.
- -ExecStart=-/sbin/agetty -o '-p -- \\u' --keep-baud 115200,38400,9600 %I $TERM
+ExecStart=-/sbin/agetty -o '-p -- \\u' --keep-baud 115200,57600,38400,9600 %I $TERM
 Type=idle
 Restart=always
 UtmpIdentifier=%I
```

**Comment by poettering** [comment]
send a PR

**Comment by mvduin** [comment]
This seems like a very bad idea. The auto-baud mechanism of getty relies on a sequence of consecutive zero-bits transmitted at a low baudrate being recognized as a break (i.e. 10 consecutive 0-bits, assuming 8 data bits and no parity or 7 data bits with parity) by the receiver configured at a high baudrate. This only works if the baudrate ratio is sufficiently large.

Switching from 38400 to 9600 requires only 3 consecutive zero-bits (transmitted at 9600 baud), while switching from 115200 to 38400 requires 4 consecutive zero-bits. Assuming we're using 8 data bits and no parity (as is typical), four consecutive zero-bits (one of which may be the start bit) can be triggered by any control character (e.g. pressing enter, escape, or ctrl-C), space, and also the `A` of modem AT-commands.

Switching from 115200 to 57600 however requires 5 consecutive zero-bits, which is still triggered by space, ctrl-C, and `A`, but no longer by pressing enter or escape. Switching from 57600 to 38400 requires **seven** consecutive zero-bits, which leaves only 5 possible bytes to trigger the switch (ctrl-@, ctrl-A, `@`, 0x80, 0xc0), none of which are likely to be seen.

In other words, in practice, using `115200,57600,38400,9600` as baudrate list makes 57600 appear to work inconsistently (since pressing enter is the most typical way to elicit a response from a terminal) and lower baudrate will not work at all.
